### PR TITLE
COMMERCE-11003 We need to filter for allowedCPDefinitionOptionValueRe…

### PR DESCRIPTION
**Pull request by brianikim**
…ls always

@andrea-ale-sbarra 

https://issues.liferay.com/browse/COMMERCE-11003

Sincerely,
Brian I. Kim

### CODE REVIEW

**Comment by brianikim** [comment]
ci:test:commerce

File: modules/apps/commerce/commerce-product-definitions-web/src/main/java/com/liferay/commerce/product/definitions/web/internal/frontend/data/set/provider/CommerceProductInstanceOptionsValuesDataProvider.java [hunk 1]
```diff
@@ -388,6 +373,39 @@ private JSONArray _addToJSONArray(
...
+				CPDefinitionOptionRel cpDefinitionOptionRel,
+				List<CPDefinitionOptionValueRel>
+					selectedCPDefinitionOptionValueRels)
+		throws PortalException {
+
+		List<CPDefinitionOptionValueRel> allowedCPDefinitionOptionValueRels =
+			_filterBySelectedCPDefinitionOptionValueRelIds(
+				cpDefinitionOptionRel, selectedCPDefinitionOptionValueRels);
+
+		if (cpDefinitionOptionRel.isPriceContributor()) {
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-definitions-web/src/main/java/com/liferay/commerce/product/definitions/web/internal/frontend/data/set/provider/CommerceProductInstanceOptionsValuesDataProvider.java:387]: I think you can add these two if in the _filterBySelectedCPDefinitionOptionValueRelIds method

**Review by andrea-ale-sbarra — CHANGES_REQUESTED** [review]

**Comment by brianikim** [comment]
ci:test:commerce

**Comment by brianikim** [comment]
@andrea-ale-sbarra , could we have QA investigate the test failures? I wonder if the tests themselves need to be changed or if there are legitimate issues that I have missed in my testing.

**Comment by brianikim** [comment]
ci:test:commerce

**Pull request closed without merging**